Feature/uf lmapping

no-reply opened the pull request:
Submitting this as a PR so there is a discussion point.

guegueng committed:
UFL mapping

beginning of mapping

Commit:
Fix syntax and name errors

Fix syntax errors.
Comment out extent mapping, with syntax error, to fix later.
Fix name of Krikri::MARCXMLParser.

Commit:
Change intermediateProvider, isShownAt, providedLabel

- Fix errors with intermediateProvider, isShownAt
- Fix error with providedLabel, but this field needs more
  work to filter values correctly. (noted in FIXME)

Commit:
Fix intermediateProvider

Commit:
Add FIXME comments and broken attempt at extent mapping [ci skip]

Commit:
UFL Now with a working extent mapping

anarchivist commented on heidrun/ufl_marc.rb:
```diff
@@ -0,0 +1,108 @@
+    end
+
+    #don't know how to do these but am keeping placeholder
+    #contributor 
```
The contributor mapping suggests that we need only `700$a` here when `$e` is not `joint author`. Do we really want to throw away the other subfields?

anarchivist commented on heidrun/ufl_marc.rb:
```diff
@@ -0,0 +1,108 @@
+    #don't know how to do these but am keeping placeholder
+    #contributor 
+
+    #creator 
```
The creator mapping suggests that we need only `$a` for 100, 110, 111 (and `700` when `$e` is `joint author` or `jt author`). Do we really want to throw away the other subfields?

anarchivist commented on heidrun/ufl_marc.rb:
```diff
@@ -0,0 +1,108 @@
+      providedLabel date.field('marc:subfield').match_attribute(:code, 'c')
+    end
+
+    #description 
```
from mapping document: `5XX; not 506, 538, 536, 535, 533, 510, 540`, so maybe something like:

``` ruby
description record.field('marc:datafield')
                  .match_attribute(:tag) { |tag| tag =~ /^5..$/ }
                  .reject { |df| ['506', '538', '536', '535', '533', '510', '540'].include? df.tag }
```

... but then we need away to concatenate the values of all of the subfields.

no-reply replied:
We should have `Array#join`. Or, at least, we could fairly easily.

no-reply replied:
I'm concerned about this mapping. The first 1000 records contain exactly one match, and it doesn't look like a description.

anarchivist commented on heidrun/ufl_marc.rb:
```diff
@@ -0,0 +1,108 @@
+
+    #language 
+
+    #spatial 
```
From mapping document:

> 752 (separate subfields with double hyphen); else 650$z; 651$a; 662 [see defs of subfield codes at http://www.loc.gov/marc/bibliographic/bd662.html]

anarchivist commented on heidrun/ufl_marc.rb:
```diff
@@ -0,0 +1,108 @@
+
+    #identifier 
+
+    #language 
```
From mapping document:

> 041$a; OR 008 (positions 35-37)

anarchivist commented on heidrun/ufl_marc.rb:
```diff
@@ -0,0 +1,108 @@
+    
+    #genre 
+
+    #identifier 
```
From mapping document:

> 001; 020 [prefix ="ISBN: "]; 022$a [prefix ="ISSN: "]; 035$a; 050$a$b [prefix ="LC call number: "]

anarchivist commented on heidrun/ufl_marc.rb:
```diff
@@ -0,0 +1,108 @@
+    rights record.field('marc:datafield').match_attribute(:tag, '506')
+                   .field('marc:subfield').match_attribute(:code, 'a')
+
+    #subject 
```
From mapping document:

> 600; 61X; 650; 651; 653; 654; 655; 656; 657; 658; 69X (separate subfield with double hyphens)

anarchivist commented on heidrun/ufl_marc.rb:
```diff
@@ -0,0 +1,108 @@
+
+    #subject 
+
+    #title 
```
From mapping document:

> 245 (all subfields except $c); 242; 240

anarchivist commented on heidrun/ufl_marc.rb:
```diff
@@ -0,0 +1,108 @@
+         :as => :date do
+      providedLabel date.field('marc:subfield').match_attribute(:code, 'c')
+    end
+
```
missing mapping (or comment) for `dcformat` - from mapping document:

> 007 position 00 [see http://www.loc.gov/marc/bibliographic/bd007.html]; position 06 in Leader [see “06 - Type of record“ here: http://www.loc.gov/marc/bibliographic/bdleader.html]; 337$a; 338$a; 340$a

anarchivist commented on heidrun/ufl_marc.rb:
```diff
@@ -0,0 +1,108 @@
+
+    #title 
+
+    #dctype
```
this should be the same thing for now as the mapping as `dcformat`

Commit:
Map contributor

Also add subfield_e method.

Commit:
Map creator

Commit:
Map description

Commit:
Add MappingTools module, Add genre mapping

Commit:
Move lambdas for selects into MappingTools module

Move lambdas that are useful in #select calls on XmlParser::Value
objects into the MappingTools module, because they're likely to be
useful across MARC mappings.

Commit:
Break up MappingTools module into separate files

Commit:
Map DC type

Commit:
Add comments

Commit:
Map title, Add element-selection methods

* Map title
* Add methods to abstract the selection of XML elements

Commit:
Fixup: Remove obsolete comment [ci skip]

Commit:
Map identifier

TODO: refactor some of the other mappings to use named lambdas.

Commit:
Use named lambdas for blocks passed to #map

Commit:
Map language

Commit:
Map spatial, Add MappingTools::MARC.all_subfield_values

Also fix and rename some existing methods that select datafields and
subfields.  Ensure that these methods work with multiple datafields of
the same number.

Commit:
Fixup: remove commented code [ci skip]

Commit:
Map subject

Commit:
Fixup comment [ci skip]

Commit:
Complete mapping of relation

Commit:
Map dcformat

Commit:
Add tests

Also:
* Explicitly call Array#first for node in
* MappingTools::MARC.subfield_values

And furthermore:
* Fix doc comments
* Remove unnecessary rescue in MappingTools::MARC.datafield_values
* Fix indentation, ufl_marc.rb

Commit:
Add .travis.yml

markbreedlove commented:
I'm closing this, and will open a new one.
